### ISSUE TITLE
Model layers are fusing between predictions  

### ISSUE THREAD

**Issue by SkalskiP** [ultralytics/ultralytics#468]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Detection

### Bug

Is it possible to prevent the model from fusing and logging?

I was testing the `YOLOv8.0.9` release. And it is still fusing and logging between predictions. Is it possible to turn that off? Or I could use the model differently to get the expected behavior.

`get_video_frames_generator(SOURCE_VIDEO_PATH)` is just returning the video frame generator. As promised, I can now use `np.ndarray` for inference, but it is still fuseing. 

![Screenshot 2023-01-18 at 14 39 41](https://user-images.githubusercontent.com/26109316/213186226-1c768404-3224-410f-a540-0030da1764c7.png)

### Environment

Google Colab

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by SkalskiP** [comment]
@AyushExel any ideas? :)

**Comment by AyushExel** [comment]
@SkalskiP I guess you're trying to do this? It doesn't happen to me.. Or maybe I completely missed your question
<img width="839" alt="Screenshot 2023-01-18 at 7 58 58 PM" src="https://user-images.githubusercontent.com/15766192/213197537-610b322b-8a8d-4644-949b-b5ef8e3a58a1.png">

**Comment by AyushExel** [comment]
@SkalskiP Oh I see what's happening. You don't have to pass video frame by frame. You can just pass the video directly and we'll return the generator if you want that. Please read first paragraph of the docs here - https://docs.ultralytics.com/predict/

**Comment by SkalskiP** [comment]
@AyushExel, but I want to pass the video frame by frame because I don't wish to YOLO to be in charge of that. That responsibility is already part of other elements in the codebase.

**Comment by glenn-jocher** [comment]
Minimum reproducible example of the issue.

```python
import cv2
import numpy as np
from ultralytics import YOLO

!wget https://github.com/ultralytics/yolov5/releases/download/v1.0/decelera_landscape.mov

model = YOLO('yolov8n.pt')
model.fuse()
 
cap = cv2.VideoCapture('decelera_landscape.mov')
while(cap.isOpened()):
  ret, frame = cap.read()
  if ret == True:
    results=model(frame)
    if cv2.waitKey(25) & 0xFF == ord('q'):
      break
  else: 
    break
 
cap.release()
cv2.destroyAllWindows()
```

**Comment by AyushExel** [comment]
@SkalskiP ok I'll fix it in main now. And should be available in tomorrow with next release

**Comment by SkalskiP** [comment]
@glenn-jocher, thanks for helping me with that! 🙏🏻 

@AyushExel great! I think it is real-life scenario for YOLOv8 python SDK.

**Comment by glenn-jocher** [comment]
@SkalskiP ok this appears resolved in #477 which will be part of 8.0.10 tonight. I tested the branch and it looks good (no screen printout anymore, no fusing):

<img width="835" alt="Screenshot 2023-01-18 at 18 29 39" src="https://user-images.githubusercontent.com/26833433/213252775-653e8e78-2a2d-4826-96c6-5f9f1ece1b6c.png">

**Comment by glenn-jocher** [comment]
@AyushExel the initial .fuse() op is redundant here though. We should have an attribute model.is_fused=True set and then not-refuse as it's a costly operation.

**Comment by AyushExel** [comment]
@glenn-jocher yeah. Fuse is called by Autobackend and not by the predictor...`is_fuse` needs to be a model attribute then

**Comment by glenn-jocher** [comment]
@AyushExel ok. Can you do that?

BTW in the future before we train we also want to 

`assert not model.is_fused, 'Fused models can not be trained, please load a new *.pt checkpoint file'

**Comment by AyushExel** [comment]
@glenn-jocher If we add the model attribute now, the released ckpts won't get it. Let's move this convo to the PR thread

**Comment by Lalita793** [comment]
How to fuse diffrent layers in every epoch that means every even and odd layer

**Comment by glenn-jocher** [comment]
@Lalita793 the standard `model.fuse()` method doesn't support selective layer fusing - it fuses all Conv2d and BatchNorm2d layers at once for inference optimization. Custom layer manipulation during training epochs would require modifying the model architecture manually, which isn't a built-in feature. Also note that fused models cannot be trained, so this approach isn't compatible with the standard training pipeline.
